`LocalDateTime` to `Instant` conversion incorrect for mingwX64 target after daylight savings in `America/New_York` on 0.6.0

In 2023, the conversion from EDT to EST happened at `2023-11-05T06:00:00Z` in `America/New_York`. This was a shift backwards of one hour in local time. On JVM, Linux, and Apple targets this conversion is calculated properly.  However, on the mingwX64 target, this time difference is improperly calculated. The following test fails on the Windows target, but passes for all other available targets. 

```kt
@Test fun calculated_instant_is_correct() {
    val tz = TimeZone.of("America/New_York")
    val expected = Instant.parse("2023-11-05T07:00:00Z")
    val actual = LocalDateTime(year = 2023, monthNumber = 11, dayOfMonth = 5, hour = 2, minute = 0).toInstant(tz)
    assertEquals(expected, actual)
}
```

With the following error message

```
 io.github.kevincianfarini.cardiologist.SuspendingTests.calculated_instant_is_correct[mingwX64] FAILED
    kotlin.AssertionError: Expected <2023-11-05T07:00:00Z>, actual <2023-11-05T06:00:00Z>.
```


In case it's useful, [here's a reference](https://www.worldtimebuddy.com/?qm=1&lid=4791259,100&h=4791259&date=2023-11-5&sln=3-3.5&hf=1) clock aligning `UTC` and `America/New_York` during the instant in question.

--- a/core/common/test/TimeZoneTest.kt
+++ b/core/common/test/TimeZoneTest.kt
@@ -281,6 +281,7 @@ private fun checkOverlap(timeZone: TimeZone, overlapStart: LocalDateTime) {
     val instantStart = overlapStart.plusNominalSeconds(-1).toInstant(timeZone).plus(1, DateTimeUnit.SECOND)
     // the later occurrence of the overlap
     val instantEnd = overlapStart.plusNominalSeconds(1).toInstant(timeZone).minus(1, DateTimeUnit.SECOND)
+    assertEquals(instantEnd, overlapStart.toInstant(timeZone))
     try {
         // there is at least a one-second overlap
         assertNotEquals(instantStart, instantEnd)

--- a/core/native/src/internal/TimeZoneRules.kt
+++ b/core/native/src/internal/TimeZoneRules.kt
@@ -167,7 +167,7 @@ internal class RecurringZoneRules(
             val ldtAfter = rule.transitionDateTime.toLocalDateTime(rule.offsetAfter)
             return if (localDateTime < ldtBefore && localDateTime < ldtAfter) {
                 OffsetInfo.Regular(rule.offsetBefore)
-            } else if (localDateTime > ldtBefore && localDateTime >= ldtAfter) {
+            } else if (localDateTime >= ldtBefore && localDateTime >= ldtAfter) {
                 offset = rule.offsetAfter
                 continue
             } else if (ldtAfter < ldtBefore) {

--- a/core/tzfile/test/TimeZoneRulesTest.kt
+++ b/core/tzfile/test/TimeZoneRulesTest.kt
@@ -22,10 +22,13 @@ class TimeZoneRulesTest {
         // on the last Sunday in October.
         val dstStartTime = LocalDateTime(2040, 3, 25, 2, 0)
         val infoAtDstStart = rules.infoAtDatetime(dstStartTime)
-        assertTrue(infoAtDstStart is OffsetInfo.Gap, "Expected Gap, got $infoAtDstStart")
+        assertIs<OffsetInfo.Gap>(infoAtDstStart)
         val dstEndTime = LocalDateTime(2040, 10, 28, 3, 0)
         val infoAtDstEnd = rules.infoAtDatetime(dstEndTime)
-        assertTrue(infoAtDstEnd is OffsetInfo.Overlap, "Expected Overlap, got $infoAtDstEnd")
+        assertIs<OffsetInfo.Regular>(infoAtDstEnd)
+        val timeBeforeDstEnd = LocalDateTime(2040, 10, 28, 2, 59, 59, 999_999_999)
+        val infoBeforeDstEnd = rules.infoAtDatetime(timeBeforeDstEnd)
+        assertIs<OffsetInfo.Overlap>(infoBeforeDstEnd)
     }
 
     @Test

